File: neatformcore/src/main/java/com/opensrp/neatformcore/domain/view/RootView.java
package com.opensrp.neatformcore.domain.view;

import com.opensrp.neatformcore.domain.model.NFormViewData;
import com.opensrp.neatformcore.domain.model.NFormViewProperty;
import com.opensrp.neatformcore.views.data.ViewDataHandler;

import java.util.List;

public interface RootView {
    RootView initRootView();

    void addChild(NFormView nFormView);

    void addChildren(List<NFormViewProperty> viewProperties, ViewDataHandler viewDataHandler);

    List<NFormViewData> getViewsData();

}

File: neat-form-core/src/main/java/com/nerdstone/neatformcore/domain/view/RootView.kt
package com.nerdstone.neatformcore.domain.view

import com.nerdstone.neatformcore.domain.builders.FormBuilder
import com.nerdstone.neatformcore.domain.model.NFormViewProperty
import java.io.Serializable

interface RootView : Serializable {

    var formBuilder: FormBuilder

    fun initRootView(formBuilder: FormBuilder): RootView

    fun addChild(nFormView: NFormView)

    fun addChildren(
        viewProperties: List<NFormViewProperty>, formBuilder: FormBuilder,
        buildFromLayout: Boolean = false
    )
}
